packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/PropertyDescriptionBuilder.cs

```diff
@@ -0,0 +1,184 @@
+            if (type.ContainsBinaryData)
+            {
+                FormattableString binaryDataExtraDescription = CreateBinaryDataExtraDescription(type, serializationFormat);
+                description = $"{description}{binaryDataExtraDescription}";
+            }
+
+            return description;
+        }
+
+        /// <summary>
+        /// This method attempts to retrieve the description for each of the union type items. For items that are lists,
+        /// the description will include details about the element type.
+        /// For items that are literals, the description will include the literal value.
+        /// </summary>
+        /// <param name="unionItems">the list of union type items.</param>
+        /// <returns>A list of FormattableString representing the description of each union item.
+        /// </returns>
+        internal static IReadOnlyList<FormattableString> GetUnionTypesDescriptions(IReadOnlyList<CSharpType> unionItems)
+        {
+            var values = new List<FormattableString>();
+
+            foreach (CSharpType item in unionItems)
+            {
+                FormattableString description;
+
+                if (item.IsLiteral && item.Literal != null)
+                {
+                    var literalValue = item.Literal;
+                    if (item.FrameworkType == typeof(string))
+                    {
+                        description = $"<description>{literalValue:L}</description>";
+                    }
+                    else
+                    {
+                        description = $"<description>{literalValue}</description>";
+                    }
+                }
+                else
+                {
+                    description = $"<description>{item:C}</description>";
+                }
+
+                values.Add(description);
+            }
+
+            return values.Distinct().ToList();
+        }
+
+        /// <summary>
+        /// This method will construct an additional description for properties that are binary data. For properties whose values are union types,
+        /// the description will include the types of values that are allowed.
+        /// </summary>
+        /// <param name="type">The CSharpType of the property.</param>
+        /// <param name="serializationFormat">The serialization format of the property.</param>
+        /// <returns>The formatted description string for the property.</returns>
+        private static FormattableString CreateBinaryDataExtraDescription(CSharpType type, SerializationFormat serializationFormat)
+        {
+            string typeSpecificDesc;
+            var unionTypes = GetUnionTypes(type);
+            IReadOnlyList<FormattableString> unionTypeDescriptions = Array.Empty<FormattableString>();
+
+            if (unionTypes.Any())
+            {
+                unionTypeDescriptions = GetUnionTypesDescriptions(unionTypes);
+            }
+
+            if (type.FrameworkType == typeof(BinaryData))
```

Comment:
we usually should not call `FrameworkType` before we check `IsFrameworkType`
How about we do:
```csharp
type.Equals(typeof(BinaryData))
```
or 
```csharp
type.WithNullable(false).Equals(typeof(BinaryData))
```
because we do not currently have a method of `EqualsIgnoreNullable` on `CSharpType`

Reply:
Given the [previous comment](https://github.com/microsoft/typespec/pull/3414#discussion_r1610620834), I will remove this class altogether from SCM library.